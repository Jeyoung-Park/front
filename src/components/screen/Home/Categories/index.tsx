import * as React from 'react';
import {Text, View, StyleSheet, FlatList} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {CategoryInterface} from '../../../../types';

interface CategoriesProps {
  categories: CategoryInterface[];
  currentCategoryId: number;
  setCurrentCategoryId: (param: number) => void;
}

const Categories = ({
  categories,
  currentCategoryId,
  setCurrentCategoryId,
}: CategoriesProps) => {
  const renderItem = ({item}: {item: CategoryInterface}) => {
    return (
      <TouchableOpacity onPress={() => setCurrentCategoryId(item.categoryId)}>
        <Text
          style={[{paddingVertical:8,}, item.categoryId === currentCategoryId && styles.highlightText]}>
          {item.categoryName}
        </Text>
      </TouchableOpacity>
    );
  };

  return (
    <FlatList
      style={styles.flatList}
      contentContainerStyle={styles.containerStyle}
      data={categories}
      keyExtractor={item => String(item.categoryId)}
      renderItem={renderItem}
    />
  );
};

export default Categories;

const styles = StyleSheet.create({
  flatList: {
    height: 50,
    width: '100%',

    marginTop: 16,
  },
  containerStyle: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  highlightText: {borderBottomColor: 'gray', borderBottomWidth: 1, fontWeight:'bold'},
});
