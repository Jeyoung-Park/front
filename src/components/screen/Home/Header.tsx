import * as React from 'react';
import {Text, View, StyleSheet, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/Feather';

interface HeaderProps {}

const Header = (props: HeaderProps) => {
  return (
    <View style={styles.container}>
      <View style={styles.button} />
      <Text>Pie</Text>
      <TouchableOpacity style={styles.button}>
        <Icon name="menu" size={24} color="black" />
      </TouchableOpacity>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 60,

    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',

    paddingHorizontal: 16,

    borderBottomColor: 'gray',
    borderBottomWidth: 1,
  },
  button: {
    width: 24,
    height: 24,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
