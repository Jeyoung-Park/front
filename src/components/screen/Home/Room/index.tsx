import * as React from 'react';
import { Text, View, StyleSheet } from 'react-native';

interface RoomProps {}

const Room = (props: RoomProps) => {
  return (
    <View style={styles.container}>
      <Text>Room</Text>
    </View>
  );
};

export default Room;

const styles = StyleSheet.create({
  container: {}
});
