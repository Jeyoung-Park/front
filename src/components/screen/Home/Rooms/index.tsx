import React from 'react';
import { Text, View, StyleSheet } from 'react-native';

interface RoomsProps {}

const Rooms = (props: RoomsProps) => {
  return (
    <View style={styles.container}>
      <Text>Rooms</Text>
    </View>
  );
};

export default Rooms;

const styles = StyleSheet.create({
  container: {}
});
