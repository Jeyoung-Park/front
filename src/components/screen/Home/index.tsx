import { useNavigation } from '@react-navigation/core';
import React, {useState} from 'react';
import {Text, View, StyleSheet, FlatList} from 'react-native';
import {CategoryInterface, RoomInterface} from '../../../types';
import Categories from './Categories';
import Rooms from './Rooms';

interface HomeProps {
  categories: CategoryInterface[];
  rooms: RoomInterface[];
}

const Home = ({categories, rooms}: HomeProps) => {

  // console.log('categories in Home, ', categories);
  // console.log('rooms in Home, ', rooms);

  const navigation=useNavigation();

  const [currentCategoryId, setCurrentCategoryId] = useState<number>(0);

  return (
    <View style={styles.container}>
      <Categories
        categories={categories}
        currentCategoryId={currentCategoryId}
        setCurrentCategoryId={param => setCurrentCategoryId(param)}
      />
      <Rooms rooms={rooms} />
    </View>
  );
};

export default Home;

const styles = StyleSheet.create({
  container: {
    flex: 1,

    paddingHorizontal: 16,
  },
});
