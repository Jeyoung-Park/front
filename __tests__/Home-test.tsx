import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../src/components/screen/Home';

describe("<Home />", ()=>{
    it('snapshot test', ()=>{
        const tree=renderer.create(
            <Home />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})