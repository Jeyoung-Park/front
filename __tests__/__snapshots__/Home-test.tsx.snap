// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> snapshot test 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "gray",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    />
    <Text>
      Pie
    </Text>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "opacity": 1,
          "width": 24,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï†œ
      </Text>
    </View>
  </View>
  <Text>
    Home
  </Text>
</View>
`;
